runtime/lua/vim/iter.lua

```diff
@@ -112,6 +112,35 @@ local function sanitize(t)
   return t
 end
 
+--- Flattens a single list-like table. Errors if it attempts to flatten a
+--- dict-like table
+---@param v table table which should be flattened
+---@param max_depth number depth to which the table should be flattened
+---@param initial_depth number|nil (internal) starting depth (defaulted to 0)
+---@param result table|nil (internal) flattened in each iteration (defaulted to {})
+---@return table flattened table if it can be flattened, otherwise nil
+local function flatten_single_list(v, max_depth, initial_depth, result)
+  local d = initial_depth or 0
+  local res = result or {}
+
+  if d < max_depth and type(v) == 'table' then
+    local i = 0
+    for _ in pairs(v) do
+      i = i + 1
+      -- short-circuit: this is not a list like table
+      if v[i] == nil then
+        error('dict-like table encountered')
+        return res
+      end
```

Comment:
```suggestion
      assert(v[i])
```

Reply:
Actually why not just ditch the error completely and return `nil` if the input is invalid. Then at the call site you can check the return value for `nil` rather than use `pcall`.

Reply:
I originally tried to do this, but I couldn't get it working in the case where the dict-like table is nested deeper than the first level that is flattened. I'd return `nil` for that one entry, but the return value of the helper function would not be `nil`. I also wanted to exit the recursion early when a dict-like table was encountered, and an exception seemed like the easiest way to do that. My mental model could totally be wrong though - this is my first time diving deep into lua 😅